androidApp/src/main/java/com/example/tasky/android/agenda/screen/AgendaScreen.kt

```diff
@@ -97,14 +108,49 @@ private fun AgendaScreen(
                 Modifier.fillMaxWidth(),
             )
             Spacer(modifier = Modifier.height(16.dp))
+            Text(
+                "Today",
+                style = typography.headlineMedium,
+                color = Black,
+                lineHeight = 16.sp,
+            )
+            Spacer(modifier = Modifier.height(16.dp))
+            LazyColumn {
+                val timeNeedleIndex = getTimeNeedleDisplayIndex(state.agendas)
+                if (timeNeedleIndex < 0) {
+                    item {
+                        TimeNeedle()
+                    }
+                }
+                itemsIndexed(state.agendas) { index, item ->
+                    AgendaCard(agendaItem = item, onOpenClick = {}, onDeleteClick = {}, onEditClick = {})
+                    TimeNeedle(modifier = Modifier.alpha(if (timeNeedleIndex == index) 1f else 0f))
+                }
+            }
         }
     }
 }
 
+private fun getTimeNeedleDisplayIndex(agendaItems: ImmutableList<AgendaItem>): Int {
+    val currentTime = System.currentTimeMillis()
+    val index = agendaItems.indexOfFirst { it.getStartTime() > currentTime }
+
+    return if (index < 0) {
+        agendaItems.lastIndex
+    } else {
+        index - 1
+    }
+}
```

Comment:
This function assumes that the list is ordered by agenda start time, and the function signature does not communicate this. Would it make more sense to update your parameter argument to clarify this assumption? I’ve found minor things like this can lead to us developers making assumptions when things aren’t clear and this can cause debugging sessions to be longer than it needs to be 😄